neural_speed/models/stablelm/stablelm.cpp

```diff
@@ -309,9 +315,13 @@ static bool stablelm_model_eval_internal(model_context* ctx, const model_input*
     {
       // Post Attention norm
       {
-        cur = ne_norm(ctx0, cur, hparams.norm_eps);
-        cur = ne_mul(ctx0, cur, model.layers[il].norm[2]);
-        cur = ne_add(ctx0, cur, model.layers[il].norm[3]);
+        if (n_layer < 40) {
+          cur = ne_norm(ctx0, cur, hparams.norm_eps);
+          cur = ne_mul(ctx0, cur, model.layers[il].norm[2]);
+          cur = ne_add(ctx0, cur, model.layers[il].norm[3]);
+        } else {
+          cur = inpPA;  // Parallel FFN
```

Comment:
why when layer is equal and large than 40, the Post Attention norm will be changed?

Reply:
seems the layer==40 has `qk norm` rather than `Post Attention Norm`.

Reply:
This models is different, it inspires from the paper "Scaling Vision Transformers to 22B". Basically they don't have post attention normalization because the MLP layer is in parallel, and its input is the same to GQA after the first LayerNorm. Check this: https://andlukyane.com/blog/paper-review-vit-22